api/v1beta1/types.go

```diff
@@ -734,6 +734,36 @@ type NetworkInterface struct {
 	// +kubebuilder:validation:nullable
 	// +optional
 	AcceleratedNetworking *bool `json:"acceleratedNetworking,omitempty"`
+
+	// Name optionally allows you to override the Azure NetworkInterface name. If left unspecified, CAPZ will
+	// generate a NetworkInterface name based on the VM name.
+	// If specified, the specified networkInterface does not exist, CAPZ will create it when the VM is created and delete it when the VM is deleted.
+	// If you specify an existing Azure NetworkInterface, CAPZ will attach and detach the interface from the VM when the VM is created and deleted,
+	// but CAPZ will not delete the NetworkInterface if the VM is deleted.
+	// +optional
+	Name *string `json:"name,omitempty"`
```

Comment:
Does this need to be a string pointer instead of a string? I would assume we can tell it's unspecified by an empty string, meaning it can't be a network interface name.